src/bsoncxx/document/value.hpp

```diff
@@ -76,6 +81,20 @@ class BSONCXX_API value {
     value(value&&) noexcept = default;
     value& operator=(value&&) noexcept = default;
 
+    // Serializer functions
+    template <typename T,
+              typename std::enable_if<!std::is_same<T, typename array::view>::value, int>::type = 0>
+    explicit value(const T& user_object)
+        : _data(new std::uint8_t[static_cast<std::size_t>(5)],
+                [](std::uint8_t* ptr) { delete[] ptr; }) {
+        to_bson(user_object, *this);
```

Comment:
So, the idea is that we would provide a to_bson(U, document::value*) method for every type U that we want to be able to serialize into a document?

Reply:
Yup! A `to_bson` serializer function should be written by the user within their object's namespace to take advantage of this